iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java

```diff
@@ -27,15 +27,18 @@
 import org.sonar.iac.docker.tree.api.ExecFormLiteralTree;
 import org.sonar.iac.docker.tree.api.ExecFormTree;
 import org.sonar.iac.docker.tree.api.SeparatedList;
+import org.sonar.iac.docker.tree.api.ShellFormTree;
 import org.sonar.iac.docker.tree.api.SyntaxToken;
 
 public class CmdTreeImpl extends InstructionTreeImpl implements CmdTree {
 
   private final ExecFormTree execForm;
+  private final ShellFormTree shellForm;
 
-  public CmdTreeImpl(SyntaxToken keyword, @Nullable ExecFormTree execForm) {
+  public CmdTreeImpl(SyntaxToken keyword, @Nullable ExecFormTree execForm, @Nullable ShellFormTree shellForm) {
     super(keyword);
     this.execForm = execForm;
+    this.shellForm = shellForm;
   }
```

Comment:
You forgot to take care of this new attribute `shellForm` in the method `cmdArguments` ;)
Also as mentioned in my other comment in `CmdTreeImplTest.java`, you don't have a test case about shellForm, so you didn't saw this miss + the resulting NullPointerExeption.

Reply:
Excellent catch!
Yes, I forgot to extend `cmdArguments()`. By the way, I also extended `CmdTree` to care `shellForm`.